IPs/RSKIPxxx.md

```diff
@@ -0,0 +1,262 @@
+
+To avoid attackers spamming the Bridge with ordinals, there are upfront and recurrent costs for parking ordinals within RSK. Ordinals whose annual fee has been unpaid will be assigned to the first user who pays its parking fees. In the future, a blinded auction process could be added.  
+ 
+The Bridge will provide these new methods:
+
+* **getFederationOrdinalsAddress**(): get the current address to transfer ordinals to Rootstock.
+* **registerBtcTransactionWithOrdinals**(): registers all the ordinals in a transaction. The ordinals are identified by being sent to the ordinals federation address.
+* **registerBtcTransactionWithOrdinalsDMFA**(): registers all ordinals in a transaction. Ordinals are identified by being sent to unique federation addresses mapped to a destination owner.
+* **getOrdinalOwner**(): Returns the RSK owner of an ordinal. 
+* **transferOrdinal**(): sets a new RSK owner for an ordinal. Only the owner can transfer ownership of  an ordinal.
+* **releaseOrdinal**(): transfers the ordinal back to a Bitcoin user of the Bitcoin blockchain. Only the owner can release an ordinal.
+* **getOrdinalsAnnualParkingFee**(): returns the fee that ordinals must pay annually for being parked. The amount is given in rBTC weis.
+* **payParkingExtensionFee**(): Pays the parking fee of an ordinal to be extended one year. 
+* **getOrdinalParkingExtensionFee**(): Returns how much the user needs to pay to extend the parking period for one year from the current date.
+* **getOrdinalParkingExpirationDate**(): Returns the date the current parking period ends.
+
+Starting from an activation block (TBD) these methods are activated.
+ 
+
+## Peg-in of an ordinal
+
+The bridge has a special address to send ordinals or inscriptions the ordFedAddress. To avoid mistakenly sending BTCs as ordinals, sending BTCs over a 0.01 BTC is forbidden. 
+
+To obtain the ordFedAddress, the following bridge method must be called:
+
+`String getFederationOrdinalsAddress()`
+
+The address is constructed differently from the BTC peg-in address. The construction is described in the section “Main Ordinals Address Derivation”
+
+To register an ordinal, the following method can be called:
+
+`void registerBtcTransactionWithOrdinals(bytes rskTx, bytes btcTxSerialized, int height, bytes pmtSerialized)`
+
+This method will register the UTXOs as independent entities. Each UTXO registered will be assigned a serial number (or index) starting from 0 on each transaction.  This serial number matches the UTXO output index. If there are outputs that are not transfers to the federation ordinal address, then those serial numbers will be skipped.
+
+The UTXOs will be owned in RSK by the same private key that controls the first input of the transaction or owned by a different RSK address, if there is an `OP_RETURN` output containing it, as specified by RSKIP-170.
```

Comment:
To make it simple, there are no rejections. Either the user includes a valid address in an OP_RETURN output or a valid P2PKH first input.
We can enable the ordinal to be claimed by anyone after 6 months. Do you think this would cover this case ?

Reply:
Regarding spamming attack: there can't be any, because the registration of the ordinal is made by the user with an RSK transaction.